### ISSUE TITLE
Further problems getting QZ2.16.38 to connect to Zwift on 240130

### ISSUE THREAD

**Issue by vineman70-3** [cagnulein/qdomyos-zwift#2075]
**Describe the bug**
I do not know. I could not get QZ to connect to Zwift, but instead it keep connecting to a Lifefitm]ness treadmill or a Samsung TV

**To Reproduce**

**Expected behavior**
I expected Zwift to see the iPad running the QZ2.16.30 app, but it keep connecting to a Lifefitness treadmill until the 9th try. Then the data is Zwift was weird with max watts at 2000!
**Screenshots**
I also attach 2 screenshots of the last 2 emails I was sent from QZ
<img width="1342" alt="0957_Ride_Screenshot 2024-01-30 at 12 01 55 PM" src="https://github.com/cagnulein/qdomyos-zwift/assets/157830113/b0fab366-e8e1-4b47-96a1-fb3778db2127">
<img width="1335" alt="1008_Run_Screenshot 2024-01-30 at 12 01 16 PM" src="https://github.com/cagnulein/qdomyos-zwift/assets/157830113/96af4c98-5087-494b-8d4e-ebf5870bc146">

**Desktop (please complete the following information):**
 - OS: [e.g. iOS]IOS1.,3 on iPad running the QZ app & same on another iPad running Zwift. 10 zip files attched.

**Smartphone (please complete the following information):**

**Append a debug log**
See the 10 atta
[ 10s.zip](https://github.com/cagnulein/qdomyos-zwift/files/14101944/10s.zip)
[4.zip](https://github.com/cagnulein/qdomyos-zwift/files/14101945/4.zip)
[5.zip](https://github.com/cagnulein/qdomyos-zwift/files/14101946/5.zip)
[6.zip](https://github.com/cagnulein/qdomyos-zwift/files/14101947/6.zip)
[7.zip](https://github.com/cagnulein/qdomyos-zwift/files/14101948/7.zip)
[8.zip](https://github.com/cagnulein/qdomyos-zwift/files/14101949/8.zip)
[9.zip](https://github.com/cagnulein/qdomyos-zwift/files/14101950/9.zip)
[240130.1.zip](https://github.com/cagnulein/qdomyos-zwift/files/14101951/240130.1.zip)
[240130.2.zip](https://github.com/cagnulein/qdomyos-zwift/files/14101952/240130.2.zip)
[240130.3.zip](https://github.com/cagnulein/qdomyos-zwift/files/14101953/240130.3.zip)
ched zip files that contain all the information sent to the "files" folder for the 10 attempts. The 9th succeeded and has the 2000watts in it.
 Follow this guide https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F
  
  
**Additional context**

**Comment by cagnulein** [comment]
Got your issue, you have the fake treadmill setting enabled in the experimental settings.

Disable it and restart qz 

Let me know

**Comment by vineman70-3** [comment]
Roberto,
 
Sorry I have been meaning to get back to you. I have been testing both the bike and the treadmill. Here is the status as far as I can tell.
Re bike: Echelon EX-7S works fine with both Zwift & Fulgaz. Watts, cadence, bpm, and inclination match between the QZ app & the other app. Elevation gain, odometer, and speed do not match, but not very important although a little annoying.
Re treadmill: LifeFitness (I do not know the model number) works OK, but it does not change the inclination nor the speed as per the Zwift app. The QZ app responds with inclination & speed when I change them manually on the treadmill. The treadmill only goes to positive inclination not negative. Is this most likely an issue with the treadmill? Here is an example of the data I get in the email it sends me from a short test yesterday. Note: my shoes have Stryd dual power meter, but as far as I know they are not paired to the QZ app so I do not know where it is getting the wet data.
Average Speed: 2.4
Max Speed: 3.1
Calories burned: 47
Distance: 0.4
Elevation Gain (feet): 23
Average Watt: 81
Max Watt: 104
Average Watt/Kg: 1.1
Max Watt/Kg: 1.4
Average Heart Rate: 100
Max Heart Rate: 140
Total Output: 54
Elapsed Time: 00:11:13
Moving Time: 00:11:13
Weight Loss (Oz): 0.21
Estimated VO2Max: 21
5 Seconds Power: 108W 1.5W/Kg
1 Minute Power: 105W 1.4W/Kg
5 Minutes Power: 99W 1.3W/Kg
Estimated FTP: -1W 
QZ version: 722 - iOS
Device: LF1B01F81D70236094
HR Device:
 
So, in general, all works OK. Thank you again.
Howard

**Comment by cagnulein** [comment]
> Elevation gain, odometer, and speed do not match, but not very important although a little annoying.

check https://robertoviola.cloud/2024/01/27/dispelling-the-myth-speed-variations-between-qz-and-zwift-are-not-a-device-issue/



> K, but it does not change the inclination nor the speed as per the Zwift app.

did you set the zwift credentials in qz? then in free run you can have auto incline



> The treadmill only goes to positive inclination not negative

does your treadmill have negative inclination? which is the min negative?



> Note: my shoes have Stryd dual power meter, but as far as I know they are not paired to the QZ app so I do not know where it is getting the wet data.

did you set under accesories, power sensor, power sensor setting?

**Comment by vineman70-3** [comment]
Will send a zip file via GitHub oftodays treadmill test session.

On Feb 14, 2024, at 8:53 AM, Roberto Viola <notifications@github.com> wrote:


Elevation gain, odometer, and speed do not match, but not very important although a little annoying.

check https://robertoviola.cloud/2024/01/27/dispelling-the-myth-speed-variations-between-qz-and-zwift-are-not-a-device-issue/

I had read that article but thought maybe there were options to match things. No real issue.
K, but it does not change the inclination nor the speed as per the Zwift app.

did you set the zwift credentials in qz? then in free run you can have auto incline

I had set those initially when you sent me the way to access 2.16.38, so been always set. I have been going to various courses in “just run” - I assume that is “free run” as opposed to a workout or erg mode. See debug file sent in GitHub.
The treadmill only goes to positive inclination not negative

does your treadmill have negative inclination? which is the min negative?

No negative inclination. Minimum inclination is 0%.
Note: my shoes have Stryd dual power meter, but as far as I know they are not paired to the QZ app so I do not know where it is getting the wet data.

did you set under accesories, power sensor, power sensor setting?

No. Power sensor is “Disabled"
[240214-treadmill.zip](https://github.com/cagnulein/qdomyos-zwift/files/14284040/240214-treadmill.zip)

**Comment by vineman70-3** [comment]
Will send a zip file via GitHub oftodays treadmill test session.

> On Feb 14, 2024, at 8:53 AM, Roberto Viola ***@***.***> wrote:
> 
> 
> Elevation gain, odometer, and speed do not match, but not very important although a little annoying.
> 
> check https://robertoviola.cloud/2024/01/27/dispelling-the-myth-speed-variations-between-qz-and-zwift-are-not-a-device-issue/
> 
I had read that article but thought maybe there were options to match things. No real issue.
> K, but it does not change the inclination nor the speed as per the Zwift app.
> 
> did you set the zwift credentials in qz? then in free run you can have auto incline
> 
I had set those initially when you sent me the way to access 2.16.38, so been always set. I have been going to various courses in “just run” - I assume that is “free run” as opposed to a workout or erg mode. See debug file sent in GitHub.
> The treadmill only goes to positive inclination not negative
> 
> does your treadmill have negative inclination? which is the min negative?
> 
No negative inclination. Minimum inclination is 0%.
> Note: my shoes have Stryd dual power meter, but as far as I know they are not paired to the QZ app so I do not know where it is getting the wet data.
> 
> did you set under accesories, power sensor, power sensor setting?
> 
No. Power sensor is “Disabled"

**Comment by cagnulein** [comment]
@vineman70-3 any news?

**Comment by vineman70-3** [comment]
Hi Roberto,

Echelon EX 7s bike works fine with the QZ app and both Zwift and FullGaz with resistance changes grade appropriate to the app. So I have been using it regularly with no problems.

However, I have been unable to get auto incline to work on the Lifefitness treadmill. So I have given up and when running on Zwift pair to my Stryd duo pods and have to change manually with the treadmill buttons for incline and speed. So never could get auto incline to work.

Thank you.
Howard

**Comment by vineman70-3** [comment]
Yes, Zwift credentialsi were set in QZ. No, none of the +/- buttons on the tiles respond including the inclination tile.

**Comment by vineman70-3** [comment]
No. I never thought to look and have no idea if any such exists. The last I tried was 2/14/2014 and I sent a zip file 240214-treadmill.zip that has 3 fit and a log file in it to GitHub back then.

Under Zwift options, aside from credentials, Zwift treadmill auto inclination is on.

Power sensor options has Power Sensor as “Stryd x”. Does it make sense to try turning on “power sensor as treadmill”. Although I assume bluetooth exists as the QZ app tiles come uo. The treadmill name is put in (to  avoid all the other Lifeness devices coming up).

**Comment by cagnulein** [comment]
For auto inclination we need to find a app that can control inclination
from Bluetooth. Did you try for example kinomap?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 9 mar 2024 alle 22:13 vineman70-3 ***@***.***>
ha scritto:

> No. I never thought to look and have no idea if any such exists. The last
> I tried was 2/14/2014 and I sent a zip file 240214-treadmill.zip that has 3
> fit and a log file in it to GitHub back then.
>
> Under Zwift options, aside from credentials, Zwift treadmill auto
> inclination is on.
>
> Power sensor options has Power Sensor as “Stryd x”. Does it make sense to
> try turning on “power sensor as treadmill”. Although I assume bluetooth
> exists as the QZ app tiles come uo. The treadmill name is put in (to avoid
> all the other Lifeness devices coming up).
>

**Comment by vineman70-3** [comment]
Hi Roberto,

I just went to the gym and could not use the app all of a sudden - it says “qdomyoszwift” Beta Has Expired. What do I need to do?

Thank you.

Best,
Howard

**Comment by cagnulein** [comment]
Anyway you can use the production one, it's up to date for this ticket

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 5 dic 2024 alle 18:06 cagnulein ***@***.***> ha
scritto:

> I sent the fresh beta today  (but also a month ago). Don't you received it?
>
>
> Roberto Viola
> Software engineer and open source enthusiast
> http://robertoviola.cloud
>
>
>
> Il giorno gio 5 dic 2024 alle 17:53 vineman70-3 ***@***.***>
> ha scritto:
>
>> Hi Roberto,
>>
>> I just went to the gym and could not use the app all of a sudden - it
>> says “qdomyoszwift” Beta Has Expired. What do I need to do?
>>
>> Thank you.
>>
>> Best,
>> Howard
>>

**Comment by vineman70-3** [comment]
Thank you, but sorry for being dense I never received anything. Should I delete the expired one from my iPad and download the current one in the Apple store? Will I loose all my settings that were working in the last working beta?

**Comment by cagnulein** [comment]
If you go directly on the store without removing it, you will not lose
anything
Let me know

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 5 dic 2024 alle 18:14 vineman70-3 ***@***.***>
ha scritto:

> Thank you, but sorry for being dense I never received anything. Should I
> delete the expired one from my iPad and download the current one in the
> Apple store? Will I loose all my settings that were working in the last
> working beta?
>
